server/v2/server.go

```diff
@@ -30,27 +34,42 @@ type HasConfig interface {
 	Config() any
 }
 
-var _ ServerModule = (*Server)(nil)
+// HasStartFlags is a server module that has start flags.
+type HasStartFlags interface {
+	StartFlags() *pflag.FlagSet
+}
+
+var _ ServerModule[transaction.Tx] = (*Server)(nil)
 
 // Configs returns a viper instance of the config file
 func ReadConfig(configPath string) (*viper.Viper, error) {
```

Comment:
Can we update the test in readconfig? create a dummy `config.toml` in testdata and verify both are read?